keras_cv/models/segmentation/deeplab.py

```diff
@@ -56,136 +71,102 @@ class DeepLabV3(tf.keras.models.Model):
-                        use_bias=False,
-                        activation="softmax",
-                        # Force the dtype of the classification head to float32 to avoid the NAN loss
-                        # issue when used with mixed precision API.
-                        dtype=tf.float32,
-                    ),
-                ]
-            )
-        self.segmentation_head = segmentation_head
-
-    def build(self, input_shape):
-        height = input_shape[1]
-        width = input_shape[2]
-        feature_map_shape = self.backbone.compute_output_shape(input_shape)
-        self.up_layer = tf.keras.layers.UpSampling2D(
-            size=(height // feature_map_shape[1], width // feature_map_shape[2]),
-            interpolation="bilinear",
+    if backbone_weights and not tf.io.gfile.exists(backbone_weights):
+        raise ValueError(
+            "The `weights` argument should be either `None` or the path to the "
+            "weights file to be loaded. Weights file not found at location: {weights}"
         )
 
-    def call(self, inputs, training=None):
-        feature_map = self.backbone(inputs, training=training)
-        output = self.aspp(feature_map, training=training)
-        output = self.up_layer(output, training=training)
-        output = self.segmentation_head(output, training=training)
-        return output
-
-    # TODO(tanzhenyu): consolidate how regularization should be applied to KerasCV.
-    def compile(self, weight_decay=0.0001, **kwargs):
-        self.weight_decay = weight_decay
-        super().compile(**kwargs)
-
-    def train_step(self, data):
-        images, y_true, sample_weight = tf.keras.utils.unpack_x_y_sample_weight(data)
-        with tf.GradientTape() as tape:
-            y_pred = self(images, training=True)
-            total_loss = self.compute_loss(images, y_true, y_pred, sample_weight)
-            reg_losses = []
-            if self.weight_decay:
-                for var in self.trainable_variables:
-                    if "bn" not in var.name:
-                        reg_losses.append(self.weight_decay * tf.nn.l2_loss(var))
-                l2_loss = tf.math.add_n(reg_losses)
-                total_loss += l2_loss
-        self.optimizer.minimize(total_loss, self.trainable_variables, tape=tape)
-        return self.compute_metrics(images, y_true, y_pred, sample_weight=sample_weight)
-
-    def get_config(self):
-        config = {
-            "classes": self.classes,
-            "backbone": self._backbone_passed,
-            "decoder": self._decoder_passed,
-            "segmentation_head": self._segmentation_head_passed,
-        }
-        base_config = super().get_config()
-        return dict(list(base_config.items()) + list(config.items()))
+    inputs = utils.parse_model_inputs(input_shape, input_tensor)
+    height = input_shape[0]
+    width = input_shape[1]
+
+    x = inputs
+
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
```

Comment:
Just to inform, model from `keras_cv.models` already take include_recaling parameters and compute it internally. Models like EfficientNet-V2 has two different rescaling approach. So, this could be an issue.

Reply:
The `include_rescaling` flag is also passed to the backbone model, so they're synced. We might want to check whether a custom backbone has a rescaling layer and raise a warning message or error if it doesn't, since it'll be easy to miss this on the user's end.

Could you clarify what you mean by EffNets having a different rescaling approach? Afaik, they just either have or don't have the layer at the start :thinking:

Reply:
About effnet-v2, [check](https://github.com/keras-team/keras/blob/db667716ac6be59a8495b58e0d2e246e9c031599/keras/applications/efficientnet_v2.py#L960-L972) in keras.applications. But it looks like, it didn't follow in [keras-cv.models](https://github.com/keras-team/keras-cv/blob/b0039e2b213474f6fca8db6c3827f0ff6d6ca6df/keras_cv/models/efficientnet_v2.py#L895-L896)

cc. @sebastian-sz

Reply:
Oh, yes. The normalization was dropped while porting them: https://github.com/keras-team/keras-cv/pull/740#discussion_r960951275

Reply:
Context: The normalization/rescaling in keras application's EfficientNetV2 is coupled with the weights ported from the original repository - they expect this kind of preprocessing.

In keras-cv the goal is to train networks from scratch so there was no need to strictly follow the same preprocessing.